tests/find_iocs_cases/urls.py

```diff
@@ -10,4 +10,36 @@
         {},
         id="URL and domains parsed",
     ),
+    param(
+        "Foo https://citizenlab.ca/about/), bar",
+        {
+            "urls": ["https://citizenlab.ca/about/"],
+        },
+        {"parse_domain_from_url": False},
+        id="URL boundary w/ ) handled properly",
+    ),
+    param(
+        "DownloadString('https://example[.]com/rdp.ps1');g $I DownloadString(\"https://example[.]com/rdp.ps2\");g $I",
+        {
+            "urls": ["https://example.com/rdp.ps1", "https://example.com/rdp.ps2"],
+        },
+        {"parse_domain_from_url": False},
+        id="URL boundary w/ single or double quotes handled properly",
+    ),
+    param(
+        "url,https://example.com/g/foo,Malicious Google Groups discussion",
+        {
+            "urls": ["https://example.com/g/foo"],
+        },
+        {"parse_domain_from_url": False},
+        id="URL boundary w/ comma handled properly",
+    ),
```

Comment:
To resolve these two failing cases, my approach will be to:

1. Identify the character before the URL
2. Either:
    - Stop parsing once that character is hit again
    - Remove that character and anything after it from the URL

Reply:
This will not handle cases like #261 where the url is not preceded by a comma (it is only postceded by one).

Reply:
I am comfortable removing punctuation from the end of a URL or it the URL is preceded by the same character, but other than that, we would have to break significantly with the RFC (which is something I need to think about more).